src/functions/questionnaire/delete-questionnaire/handler.ts

```diff
@@ -0,0 +1,50 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyHandler } from "aws-lambda";
+import { questionnaireService } from "src/database/services";
+
+/**
+ * Lambda for deleting a questionnaire entry from DynamoDB.
+ * 
+ * @param event event
+ */
+
+const deleteQuestionnaireHandler: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {
+	const { id } = event.pathParameters || {};
+
+	try {
+    if (!id) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({
+          error: "Missing or invalid path parameter: id",
+        }),
+      };
+    }
+
+		const findQuestionnaireById = await questionnaireService.findQuestionnaire(id);
+		if (!findQuestionnaireById) {
+			return {
+				statusCode: 404,
+				body: JSON.stringify({
+					error: "Questionnaire not found.",
+				}),
+			};
+		};
+
+    await questionnaireService.deleteQuestionnaire(id);
+
+    return {
+      statusCode: 204,
+      body: JSON.stringify("Questionnaire deleted successfully."),
```

Comment:
Typical REST conventions for deletion is to return a status 204, which is correct, and just an empty body. So it in this you can just return:
body: ""